packages/modules/order/src/services/order-module-service.ts

```diff
@@ -2306,106 +2342,144 @@ export default class OrderModuleService<
+      {
+        id: transactionIds,
+      },
+      {
+        select: ["order_id", "amount"],
+      },
+      sharedContext
+    )
 
-    trxs = await this.updateTransactionsWithSelector_(toUpdate, sharedContext)
+    const returned = await this.transactionService_.softDelete(
+      transactionIds,
+      sharedContext
+    )
 
-    return await this.baseRepository_.serialize<
-      OrderTypes.OrderTransactionDTO[]
-    >(trxs, {
-      populate: true,
-    })
+    await this.updateOrderPaidRefundableAmount_(
+      transactions,
+      true,
+      sharedContext
+    )
+
+    return returned as any
   }
 
   @InjectTransactionManager("baseRepository_")
-  protected async updateTransaction_(
-    trxId: string,
-    data: Partial<OrderTypes.UpdateOrderTransactionDTO>,
-    @MedusaContext() sharedContext: Context = {}
-  ): Promise<Transaction> {
-    const [trx] = await this.transactionService_.update(
-      [{ id: trxId, ...data }],
+  // @ts-ignore
+  public async restoreTransactions<TReturnableLinkableKeys extends string>(
+    transactionIds: string[],
+    config?: RestoreReturn<TReturnableLinkableKeys>,
+    sharedContext?: Context
+  ): Promise<Record<string, string[]> | void> {
+    const transactions = await this.transactionService_.list(
+      {
+        id: transactionIds,
+      },
+      {
+        select: ["order_id", "amount"],
+        withDeleted: true,
+      },
       sharedContext
     )
 
-    return trx
+    const returned = await this.transactionService_.restore(
+      transactionIds,
+      sharedContext
+    )
+
+    await this.updateOrderPaidRefundableAmount_(
+      transactions,
+      false,
+      sharedContext
+    )
+
+    return returned as any
   }
 
   @InjectTransactionManager("baseRepository_")
-  protected async updateTransactionsWithSelector_(
-    updates: OrderTypes.UpdateOrderTransactionWithSelectorDTO[],
-    @MedusaContext() sharedContext: Context = {}
-  ): Promise<Transaction[]> {
-    let toUpdate: UpdateOrderTransactionDTO[] = []
+  private async updateOrderPaidRefundableAmount_(
+    transactionData: {
+      order_id: string
+      amount: BigNumber | number | BigNumberInput
+    }[],
+    isRemoved: boolean,
+    sharedContext?: Context
+  ) {
+    const summaries: any = await super.listOrderSummaries(
+      {
+        order_id: transactionData.map((trx) => trx.order_id),
+      },
+      {},
+      sharedContext
+    )
 
-    for (const { selector, data } of updates) {
-      const trxs = await super.listTransactions(
-        { ...selector },
-        {},
-        sharedContext
+    summaries.forEach((summary) => {
+      let trxs = transactionData.filter(
+        (trx) => trx.order_id === summary.order_id
       )
 
-      trxs.forEach((trx) => {
-        toUpdate.push({
-          ...data,
-          id: trx.id,
-        })
-      })
-    }
+      if (!trxs.length) {
+        return
+      }
+      transformPropertiesToBigNumber(trxs)
```

Comment:
I thought conversion to BN happens automatically?

Reply:
This property `totals` is a generic JSONB object, so the conversion won't happen automatically as it does when the entity is a Big Number instance on the model.

Reply:
Got it 👍